src/sc/sidechainrpc.cpp

```diff
@@ -440,6 +443,21 @@ bool AddSidechainCreationOutputs(UniValue& sc_crs, CMutableTransaction& rawTx, s
         uint256 address;
         address.SetHex(inputString);
 
+        const UniValue& certProvingSystemValue = find_value(o, "certProvingSystem");
+        if (certProvingSystemValue.isNull() || !certProvingSystemValue.isNum())
+        {
+            error = "Invalid parameter or missing certProvingSystem key";
+            return false;
+        }
+        uint8_t certProvingSystemIndex = certProvingSystemValue.get_int();
+        if (certProvingSystemIndex < static_cast<uint8_t>(Sidechain::ProvingSystemType::CoboundaryMarlin) ||
```

Comment:
Consider using an exact match for all supported enum values instead of a range check. The same applies also for analogous check below.